Source/Dafny/AST/DafnyAst.cs

```diff
@@ -56,28 +56,35 @@ public interface IToken : Microsoft.Boogie.IToken {
     string LeadingTrivia { get; set; }
   }
 
+  /// <summary>
+  /// Has one-indexed line and column fields
+  /// </summary>
   public record Token : IToken {
     public Token peekedTokens; // Used only internally by Coco when the scanner "peeks" tokens. Normallly null at the end of parsing
     public static readonly IToken NoToken = (IToken)new Token();
 
-    public Token() : this(0, 0) {
-    }
+    public Token() : this(0, 0) { }
 
     public Token(int linenum, int colnum) {
       this.line = linenum;
       this.col = colnum;
-      this.val = "anything so that it is nonnull";
+      this.val = "";
```

Comment:
I'm fine with this, but note that having this odd value is somewhat nice, beause it lets you see clearly that this is the default value

Reply:
An empty val, even if given a correct line and column, might result in no error displayed in VSCode. Can you please double check?